lale/lib/lale/group_by.py

```diff
@@ -12,16 +12,75 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import ast
+
+import numpy as np
+import pandas as pd
+
 import lale.docstrings
 import lale.operators
 
+try:
+    from pyspark.sql.dataframe import DataFrame as spark_df
+
+    spark_installed = True
+
+except ImportError:
+    spark_installed = False
+
+
+def _is_pandas_df(df):
+    return isinstance(df, pd.DataFrame) or isinstance(df, pd.Series)
+
+
+def _is_spark_df(df):
+    return isinstance(df, spark_df)
```

Comment:
same issue as before.  Also, maybe these methods should move to some helper file so that they are not repeated twice?